miio/waterpurifier_yunmi.py

```diff
@@ -0,0 +1,236 @@
+import logging
+from typing import Any, Dict
+
+from .click_common import command, format_output
+from .device import Device
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WaterPurifierYunmiStatus:
+    """Container for status reports from the water purifier (Yunmi model)."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Status of a Water Purifier C1 (yummi.waterpuri.lx11):
+            [0, 7200, 8640, 520, 379, 7200, 17280, 2110, 4544,
+            80, 4, 0, 31, 100, 7200, 8640, 1440, 3313]
+
+        Parsed by WaterPurifierYunmi device as:
+            {'run_status': 0, 'f1_totalflow': 7200, 'f1_totaltime': 8640,
+             'f1_usedflow': 520, 'f1_usedtime': 379, 'f2_totalflow': 7200,
+             'f2_totaltime': 17280, 'f2_usedflow': 2110, 'f2_usedtime': 4544,
+             'tds_in': 80, 'tds_out': 4, 'rinse': 0, 'temperature': 31,
+             'tds_warn_thd': 100, 'f3_totalflow': 7200, 'f3_totaltime': 8640,
+             'f3_usedflow': 1440, 'f3_usedtime': 3313}
+        """
+        self.data = data
+
+    @property
+    def run_status(self) -> int:
+        """Current running status, 0 for no error."""
+        return self.data["run_status"]
```

Comment:
Found some tips in Mi Home plugin.
```Javascript
function getError(errorCode) {
  var errorCode = Number(errorCode).toString(2);
  var len = 32;
  errorCode = Base.paddingZero(errorCode, len);
  var errors = Base.lang.exceptionsArr1;

  // ...

  var curError = [];

  for (var i = len - 1; i >= 0; i--) {
    if (errorCode.charAt(i) === '1') {
      var index = len - 1 - i;
      var target = errors[index];

      if (target && target.length > 0) {
        curError.push({
          index: index,
          name: target[0],
          detail: target[1] || '',
          type: 'error'
        });
      }
    }
  }

  return curError;
}
```
Seems the error code is able to represent multiple errors at a time by binary bit.
And the error code description is like 
```Javascript
[
	['Temperature anomaly', "1. Check if inlet water temperature is among 5~38°C, if not, please adjust the water temperature and try to purify water for several times to check if the problem is solved; \n\n2. If the problem still remains, go to www.mi.com to learn more."],
	['Meter damaged', "1. After disassembling the filter for several times, try to purify water again to see if the problem can be solved;\n\n2. If the problem still remains, go to www.mi.com to learn more."],
	['Flow sensor anomaly', '1. Disassemble the touch faucet as per manual book and open the water valve to check if the water flow is too small.'],
	['Filter life expired', 'Replace filter'],
	['Network connection error', '1. The problem may caused by Wi-Fi module damage or abnormal connection to motherboard, go to www.mi.com to learn more.'],
	['EEPROM communication error', '1. The problem may caused by EEPROM chip damage or abnormal connection to motherboard, go to www.mi.com to learn more.'],
	['RFID communication error', 'RFID communication error'],
	['Faucet communication error', '1. Water in faucet internal circuit board can cause failure, please plug device in again to check if indicator lights and keys work normal;'],
	['Purified water flow error', '1. Check if all 4 filters are properly installed and replace the filter if it expires;\n\n2. Disassemble the touch faucet as per manual book and open the water valve to check if the water flow is too small; '],
	['Water leak', 'Water leak'],
	['Floater error', 'Floater error'],
	['TDS error', "1. If the RO filter expired, replace it first and try again;\n\n2. Inlet water should be municipal water, if it's not, replace the inlet water and try again;\n\n3. If the problem still remains, go to www.mi.com to learn more."],
	['Water temperature too high', "1. Check if inlet water is warm water with temperature ≥40°C. If the temperature ≥40°C, please switch to cooling mode until the temperature < 40°C first, and try to purify the water several times; \n\n2. If the problem still remains, go to www.mi.com to learn more."],
	['Recovery rate error', "1. Check if concentrated water's waste pipe works normally, if it's bent, squashed or clogged;\n\n2. If the RO filter expired, replace it first and try again;\n\n3. If the problem still remains, go to www.mi.com to learn more."],
	["Water quality anomaly','1. Check if concentrated water's waste pipe works normally, if it's bent, squashed or clogged;\n\n2. If the RO filter expired, replace it first and try again;\n\n3. If the problem still remains, go to www.mi.com to learn more."],
	['Thermal protection for pumps', 'Thermal protection for pumps'],
	['干烧保护', '1.制常温水。\n2.再次启动制热'],
	['出水NTC异常', '1.断电重启。\n2.请联系400售后维修服务'],
	['干烧NTC异常', '1.断电重启。\n2.请联系400售后维修服务'],
	['加热器异常', '1.断电重启。\n2.请联系400售后维修服务']
]
```
Should I show both the error name and the operation advice, or just the error name to avoid being too verbose?

Reply:
The errors are reported as enums, so it should just show the "error name". The name should be informative enough to give a clue how to make it clear that it's error on the device and a pointer towards how to fix it :-) Take a look at how some other compatible devices are reporting their errors.

Reply:
Do you agree with the current implementation of the errors at the OperationStatus class? I assume an Enum doesn't work here because of a bitmask of errors / multiple errors.